src/modules/Dashboard/features/Applet/Activities/Activities.tsx

```diff
@@ -47,17 +38,19 @@ export const Activities = () => {
     () => setIsLoading(false),
   );
 
-  const { handleSearch, handleReload, searchValue } = useTable(async ({ params }) => {
+  useEffect(() => {
     if (!appletId) return;
 
-    setIsLoading(true);
-    await getActivities({
+    getActivities({
       params: {
-        ...params,
         appletId,
       },
     });
-  });
+
+    return () => {
+      setActivitiesData(null);
+    };
+  }, [appletId, getActivities]);
 
   const actions = useMemo(
```

Comment:
Would it be possible/practical to move this memo, and maybe the `formatRow` callback as well, into a shared hook that lives in the `ActivityGrid` component folder? (Like `ActivityGrid.hooks.tsx`?) Then the consuming screen could still override one of the `actions` if needed.